### PR TITLE
Check instance exists before getting status during provisioning

### PR DESCRIPTION
**Pull request by A-Beck**
During AWS provisioning, I have been experiencing  intermittent errors indicating invalid instance IDs ( longer trace attached ):

`MIQ(ManageIQ::Providers::Amazon::CloudManager::Provision#provision_error) [[Aws::EC2::Errors::InvalidInstanceIDNotFound]: The instance ID 'i-xxxxxxxxxxxx' does not exist] encountered during phase [poll_clone_complete]`

These requests did not include additional network interfaces. The solution posted to a similar issue - https://access.redhat.com/solutions/3638601 - did not help in my case.

This appears to be because the clone_task_check did not check if the instance existed before getting instance status. Since implementing this PR in my environment, I have not encountered errors during AWS provisioning.

[aws-prov-trace-20200228-1258.log](https://github.com/ManageIQ/manageiq-providers-amazon/files/4299432/aws-prov-trace-20200228-1258.log)

### CODE REVIEW

**Comment by djberg96** [comment]
👍

File: app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb [hunk 1]
```diff
@@ -2,9 +2,13 @@ module ManageIQ::Providers::Amazon::CloudManager::Provision::Cloning
   def do_clone_task_check(clone_task_ref)
     source.with_provider_connection(:sdk_v2 => true) do |ec2|
       instance = ec2.instance(clone_task_ref)
-      status   = instance.state.name.to_sym
-      return true if status == :running
-      return false, status
+      if instance.exists?
```
Comment (agrare) [inline app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb]:
```
return false unless instance.exists?

status = instance.state.name.to_sym
return true if status == :running
return false, status
```

would look a little cleaner

File: app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb [hunk 2]
```diff
@@ -2,9 +2,13 @@ module ManageIQ::Providers::Amazon::CloudManager::Provision::Cloning
...
+        status   = instance.state.name.to_sym
```
Comment (agrare) [inline app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb]: Don't need the extra padding if this isn't following https://github.com/ManageIQ/manageiq-providers-amazon/pull/604/files#diff-d41a9db602874dbc463a106f2cd10f97R4

File: app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb (hunk 1 above)
Comment (A-Beck) [reply app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb]: That does look much cleaner - thanks for the suggestion! I will update the PR.

I still want to return a state, just so the log message here will continue to be useful.

https://github.com/ManageIQ/manageiq/blob/413d9e2c174052435e8c61834f626000fbeff83c/app/models/manageiq/providers/cloud_manager/provision/state_machine.rb#L46

Comment (agrare) [reply app/models/manageiq/providers/amazon/cloud_manager/provision/cloning.rb]: :+1: You're right `return false, :pending if yadda yadda`  then

**Comment by agrare** [comment]
Thanks @A-Beck !

**Comment by agrare** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1830305

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 0ef7a2c2611723eab527ebf04fad3747f0168beb
Author: Adam Grare <agrare@redhat.com>
Date:   Fri Mar 6 13:14:32 2020 -0500

    Merge pull request #604 from A-Beck/check-instance-existance-before-getting-status

    Check instance exists before getting status during provisioning

    (cherry picked from commit 262d51c7ce5d41e061960618b816255bf5282e8b)

    https://bugzilla.redhat.com/show_bug.cgi?id=1830305
```

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit 3ddfa00cddfa93722d33801fc7c80bfd433ffb4d
Author: Adam Grare <agrare@redhat.com>
Date:   Fri Mar 6 13:14:32 2020 -0500

    Merge pull request #604 from A-Beck/check-instance-existance-before-getting-status

    Check instance exists before getting status during provisioning

    (cherry picked from commit 262d51c7ce5d41e061960618b816255bf5282e8b)

    https://bugzilla.redhat.com/show_bug.cgi?id=1830305
```

**Pull request merged**